pkg/simple/client/monitoring/prometheus/promql.go

```diff
@@ -365,17 +360,56 @@ func makePodMetricExpr(tmpl string, o monitoring.QueryOptions) string {
 		} else {
 			podSelector = fmt.Sprintf(`pod=~"%s", namespace="%s"`, o.ResourceFilter, o.NamespaceName)
 		}
-	}
-
-	// For monitoring pods on the specific node
-	// GET /nodes/{node}/pods/{pod}
-	if o.NodeName != "" {
-		if o.PodName != "" {
-			podSelector = fmt.Sprintf(`pod="%s", node="%s"`, o.PodName, o.NodeName)
+	} else {
+		var namespaces, pods []string
+		if o.NamespacedResourcesFilter != "" {
+			for _, np := range strings.Split(o.NamespacedResourcesFilter, "|") {
+				if nparr := strings.SplitN(np, "/", 2); len(nparr) > 1 {
+					namespaces = append(namespaces, nparr[0])
+					pods = append(pods, nparr[1])
+				} else {
+					pods = append(pods, np)
+				}
+			}
+		}
+		// For monitoring pods on the specific node
+		// GET /nodes/{node}/pods/{pod}
+		// GET /nodes/{node}/pods
+		if o.NodeName != "" {
+			if o.PodName != "" {
+				if nparr := strings.SplitN(o.PodName, "/", 2); len(nparr) > 1 {
+					podSelector = fmt.Sprintf(`namespace="%s",pod="%s", node="%s"`, nparr[0], nparr[1], o.NodeName)
+				} else {
+					podSelector = fmt.Sprintf(`pod="%s", node="%s"`, o.PodName, o.NodeName)
+				}
+			} else {
+				resourcesFilter := o.ResourceFilter
+				if len(pods) > 0 {
+					resourcesFilter = fmt.Sprintf("(%s)|(%s)", o.ResourceFilter, strings.Join(pods, "|"))
+				}
+				if len(namespaces) == 0 {
+					podSelector = fmt.Sprintf(`pod=~"%s", node="%s"`, resourcesFilter, o.NodeName)
+				} else {
+					podSelector = fmt.Sprintf(`namespace=~"%s", pod=~"%s", node="%s"`,
+						strings.Join(namespaces, "|"), resourcesFilter, o.NodeName)
+				}
+			}
 		} else {
-			podSelector = fmt.Sprintf(`pod=~"%s", node="%s"`, o.ResourceFilter, o.NodeName)
+			// For monitoring pods in the whole cluster
+			// Get /pods
+			resourcesFilter := o.ResourceFilter
+			if len(pods) > 0 {
+				resourcesFilter = fmt.Sprintf("(%s)|(%s)", resourcesFilter, strings.Join(pods, "|"))
```

Comment:
/pods API is newly added, do we still need to add the content of old resourcesFilter to the content of namespacedResourceFilter ?

It's ok to do so for /nodes/{node}/pods API for backward-compatible.